src/kartograf/atom_mapper.py

```diff
@@ -855,6 +854,38 @@ def suggest_mapping_from_rdmols(
 
         return mapping
 
+    def _split_protein_components_molecules(self, protein: ProteinComponent) -> list[ProteinComponent]:
+        """
+        Aims at splitting a protein component into different protein components based on the
+        connectivity of the molecules that compose it. Useful for mapping multimer components
+        or proteins with structural waters or similarly.
+
+        This returns a ``ProteinComponent`` object with the name attribute indicating the starting
+        index in the original component.
+        """
+        from rdkit.Chem.rdmolops import GetMolFrags
+        rdmol = protein.to_rdkit()
+        fragments_indices = GetMolFrags(rdmol)
+        components = []
+        for index_tuple in fragments_indices:
+            edit_rdmol_frag = Chem.EditableMol(rdmol)
+            remove_indices = []
+            for atom in rdmol.GetAtoms():
+                atom_index = atom.GetIdx()
+                if not (atom_index in index_tuple):
+                    remove_indices.append(atom_index)
+            # Need to remove separately https://github.com/rdkit/rdkit/issues/1366
+            for atom_idx in sorted(remove_indices, reverse=True):
+                edit_rdmol_frag.RemoveAtom(atom_idx)
+            #  Create component with the remaining molecule
+            frag_rdmol = edit_rdmol_frag.GetMol()
```

Comment:
Do we need to do anything about bond orders? I.e. do we know if removing the atoms also re-adjusts the bonds in the molecule?

Reply:
Maybe a test that checks the bond orders fror the components being returned would be useful?